climada/hazard/trop_cyclone.py

```diff
@@ -214,12 +215,14 @@ def set_from_tracks(self, tracks, centroids=None, description='',
                 if perc - last_perc >= 10:
                     LOGGER.info("Progress: %d%%", perc)
                     last_perc = perc
-                self.append(
+                tc_haz.append(
                     self._tc_from_track(track, centroids, coastal_idx,
                                         model=model, store_windfields=store_windfields,
                                         metric=metric))
             if last_perc < 100:
                 LOGGER.info("Progress: 100%")
+        LOGGER.debug('Append events.')
+        self.__dict__ = TropCyclone.concat([self] + tc_haz).__dict__
```

Comment:
Without this line, the function is broken for people that use `pool`.

Reply:
But replacing the `__dict__` of the object is also quite sub-optimal. Can one do without it?

Reply:
How do you mean it's sub-optimal? What's wrong with it? There are already several places in CLIMADA with this kind of assignment to `self.__dict__`.

Reply:
Having it in older code doesn't make it optimal :D . More seriously, first of all it is not recommend to do so: https://docs.python.org/3/library/stdtypes.html 
```
A special attribute of every module is __dict__. This is the dictionary containing the module’s symbol table. Modifying this dictionary will actually change the module’s symbol table, but direct assignment to the __dict__ attribute is not possible (you can write m.__dict__['a'] = 1, which defines m.a to be 1, but you can’t write m.__dict__ = {}). Modifying __dict__ directly is not recommended.
```
I think what would be better is  to have a proper way to initialize the object and then use this. Here I would like to ask @emanuel-schmid to give his input, as he can do that much better than I could.

Reply:
No, that quotation clearly refers to the `__dict__` attribute of Python modules. Here, we have a class instance's `__dict__` attribute.